lib/cocoapods/target/pod_target.rb

```diff
@@ -188,8 +188,19 @@ def dependencies
     #         dependency upon.
     #
     def recursive_dependent_targets
-      targets = dependent_targets + dependent_targets.flat_map(&:recursive_dependent_targets)
-      targets.uniq!
+      targets = dependent_targets.clone
+
+      i = 0
+      while i < targets.length
```

Comment:
@segiddins I thought you couldn't mutate an array while iterating over it, turns out you can! I'll change it